### ISSUE TITLE
libuct_ib.so.0: undefined symbol: uct_ib_exp_qp_fill_attr

### ISSUE THREAD

**Issue by kcgthb** [openucx/ucx#4883]
Hi!

I'm trying to install UCX 1.7.0 and although compilation doesn't report any error, I got an error about an undefined symbol when running `ucx_perftest` or even `ucx_info`:

```
$ ucx_info -d -t rc
#
# Memory domain: mlx5_0
#     Component: ib
#             register: unlimited, cost: 90 nsec
#           remote key: 8 bytes
#           local memory handle is required for zcopy
#
#   Transport: rc
#      Device: mlx5_0:1
ucx_info: symbol lookup error: /share/software/user/open/ucx/1.7.0/lib/ucx/libuct_ib.so.0: undefined symbol: uct_ib_exp_qp_fill_attr
```

I just tried version 1.6.0 on the same machine, with the exact same compilation environment, and `ucx_info` doesn't generate any symbol error.

Is it a know issue?

Thanks!

**Comment by yosefe** [comment]
@kcgthb probably UCX was compiled with MLNX_OFED legacy libs, but ran on a machine with MLNX_OFED upstream libs or Inbox drivers

**Comment by kcgthb** [comment]
@yosefe thanks for the suggestion!

Unfortunately, the error actually happens on the same machine UCX was compiled on, right from the building directory:

```
$ make clean
$ ../configure 
[...]
configure: =========================================================
configure: UCX build configuration:
configure:       Build prefix:   /usr
configure: Preprocessor flags:   -DCPU_FLAGS="" -I${abs_top_srcdir}/src -I${abs_top_builddir} -I${abs_top_builddir}/src
configure:         C compiler:   gcc -O3 -g -Wall -Werror -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unused-label -Wno-long-long -Wno-endif-labels -Wno-sign-compare -Wno-multichar -Wno-deprecated-declarations -Winvalid-pch -Wno-pointer-sign -Werror-implicit-function-declaration -Wno-format-zero-length
configure:       C++ compiler:   g++ -O3 -g -Wall -Werror -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unused-label -Wno-long-long -Wno-endif-labels -Wno-sign-compare -Wno-multichar -Wno-deprecated-declarations -Winvalid-pch
configure:       Multi-thread:   disabled
configure:          MPI tests:   disabled
configure:      Devel headers:   no
configure:           Bindings:   < >
configure:        UCT modules:   < ib cma >
configure:       CUDA modules:   < >
configure:       ROCM modules:   < >
configure:         IB modules:   < >
configure:        UCM modules:   < >
configure:       Perf modules:   < >
configure: =========================================================
$ make -j
[...]
$ ./src/tools/info/ucx_info -d -t rc_verbs
#
# Connection manager: tcp
#      max_conn_priv: 2040 bytes
#
# Memory domain: mlx5_0
#     Component: ib
#             register: unlimited, cost: 180 nsec
#           remote key: 8 bytes
#           local memory handle is required for zcopy
#
#   Transport: rc_verbs
#      Device: mlx5_0:1
/dev/shm/ucx/src/tools/info/.libs/lt-ucx_info: symbol lookup error: /dev/shm/ucx/src/ucs/.libs/ucx/libuct_ib.so.0: undefined symbol: uct_ib_exp_qp_fill_attr
$
``` 

So it doesn't look like a lib mismatch between building and running host.

**Comment by yosefe** [comment]
@kcgthb what is the output of `rpm -q libibverbs` and `ofed_info -s` ?
can you pls also upload the full config.log file?

**Comment by kcgthb** [comment]
@yosefe Sure!
```
$ rpm -q libibverbs
libibverbs-41mlnx1-OFED.4.7.0.0.2.47100.x86_64
$ ofed_info -s
OFED-internal-4.7-1.0.0:
```

And here's the [`config.log`](https://github.com/openucx/ucx/files/4340571/config.log) file.

**Comment by Artemy-Mellanox** [comment]
@kcgthb OFED-internal-4.7-1.0.0 is early development version, not fully UCX compatible.
Could you please upgrade to official version from  [MLNX_OFED](https://www.mellanox.com/products/infiniband-drivers/linux/mlnx_ofed)

**Comment by kcgthb** [comment]
Oh ok. Our MOFED installation comes directly from the Mellanox RPM repository at 
http://linux.mellanox.com/public/repo/mlnx_ofed/4.7-1.0.0.1/, esp. `libibverbs-41mlnx1-OFED.4.7.0.0.2.47100.x86_64` and `ofed-scripts-4.7-OFED.4.7.1.0.0.x86_64` (which provides `ofed_script` both come from http://linux.mellanox.com/public/repo/mlnx_ofed/4.7-1.0.0.1/rhel7.6/x86_64/MLNX_LIBS/

Does that mean this repository should not be used? What version should be used instead?

Thanks!

**Comment by Artemy-Mellanox** [comment]
Looks like your setup missing module  [libmlx5-devel](http://linux.mellanox.com/public/repo/mlnx_ofed/4.7-1.0.0.1/rhel7.6/x86_64/MLNX_LIBS/libmlx5-devel-41mlnx1-OFED.4.7.0.3.3.47100.x86_64.rpm), installing and automake/rebuild UCX should be sufficient.

I would recommend latest version  [MOFED 4.7-3.2.9](http://linux.mellanox.com/public/repo/mlnx_ofed/4.7-3.2.9.0/rhel7.6/x86_64/MLNX_LIBS/), though.

**Comment by kcgthb** [comment]
> Looks like your setup missing module libmlx5-devel, installing and configure/make UCX should be sufficient.

Ah yes, that did the trick, thank you!

> I would recommend latest version MOFED 4.7-3.2.9, though.

Thanks for the pointer, we'll consider upgrading our environment to that version.

Is 5.0-1.0.0.0 worth considering at this time, or should it be considered an early development verion, like 4.7-1.0.0 was?

**Comment by yosefe** [comment]
@kcgthb MLNX_OFED 5.0 is GA level and recommended to use, it was tested and certified with latest UCX (v1.8 which is now in rc1 and will be released soon).
In any case, @Artemy-Mellanox also fixed UCX to handle the intermediate MLNX_OFED version. @Artemy-Mellanox can you pls like to the fix PR?

**Comment by yosefe** [comment]
fixed in #4900
